coresdk/src/coresdk/quad_geometry.cpp

```diff
@@ -24,3 +26,35 @@ quad quad_from(float x_top_left,float y_top_left,float x_top_right,float y_top_r
     result.points[3].y = y_bottom_right;
     return result;
 }
+
+
+
+quad quad_from(const rectangle &rect)
+{
+    quad result;
+    result = quad_from(
+                       rectangle_left(rect), rectangle_top(rect),
+                       rectangle_right(rect), rectangle_top(rect),
+                       rectangle_left(rect), rectangle_bottom(rect),
+                       rectangle_right(rect), rectangle_bottom(rect) );
+    return result;
+}
+
+point_2d matrix_multiply(const matrix_2d &m, const point_2d &pts)
```

Comment:
Lets consolidate the matrix code in the matrix_2d files... move this function to the matrix_2d.cpp file ( and the associated header changes as well )